### DISCUSSION TITLE
Indexing only new documents in server mode

### DISCUSSION THREAD

**Discussion by erikkri** [ICIJ/datashare#919] [Q&A] [answered]
SSIA really: is it possible to index only new documents in server mode?
The docs only says to first SCAN then INDEX, but this seems to re-index everything.

**Comment by mvanzalu** [comment]
We created an issue about it [here](#925 )

**Comment by mvanzalu** [comment, marked as answer]
Hello @erikkri, yes it is possible to do so.
After adding your new documents in your folder, you will have to execute two commands : 

- `datashare -m CLI  --defaultProject nameOfYourIndex --stages "SCANIDX"  --reportName nameOfYourQueue` (This command will scan your index and gather the paths to store it inside the report queue)
- `datashare -m CLI --dataDir path/to/your/directory --defaultProject nameOfYourIndex --stages "SCAN,INDEX"  --reportName nameOfYourQueue`(This command will scan and index the documents in your directory, thanks to the report queue, it will skip the paths inside of it)
